tests/imfile-endmsg.regex-with-example.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+# This is part of the rsyslog testbench, licensed under ASL 2.0
+# This test tests imfile endmsg.regex.
+echo ======================================================================
+echo [imfile-endmsg.regex.sh]
+. $srcdir/diag.sh check-inotify
+. $srcdir/diag.sh init
+generate_conf
+add_conf '
+module(load="../plugins/imfile/.libs/imfile")
+module(load="../plugins/mmnormalize/.libs/mmnormalize")
+
+input(type="imfile"
+      File="./'$RSYSLOG_DYNNAME'.*.input"
+      Tag="file:" addMetadata="on" escapelf="off"
+      endmsg.regex="(^[^ ]+ (stdout|stderr) F )|(\\n\"}$)")
+
+if $msg startswith "{" then {
+	action(type="mmnormalize" rulebase="'$srcdir/imfile-endmsg.regex.json.rulebase'")
+	foreach ($.ii in $!multilinejson) do {
+		if strlen($!@timestamp) == 0 then {
+			set $!@timestamp = $.ii!time;
+		}
+		if strlen($!stream) == 0 then {
+			set $!stream = $.ii!stream;
+		}
+		if strlen($!log) == 0 then {
+			set $!log = $.ii!log;
+		} else {
+			reset $!log = $!log & $.ii!log;
+		}
+	}
+	unset $!multilinejson;
+} else {
+	action(type="mmnormalize" rulebase="'$srcdir/imfile-endmsg.regex.crio.rulebase'")
+	foreach ($.ii in $!multilinecrio) do {
+		if strlen($!@timestamp) == 0 then {
+			set $!@timestamp = $.ii!time;
+		}
+		if strlen($!stream) == 0 then {
+			set $!stream = $.ii!stream;
+		}
+		if strlen($!log) == 0 then {
+			set $!log = $.ii!log;
+		} else {
+			reset $!log = $!log & $.ii!log;
+		}
+	}
+	unset $!multilinecrio;
+}
+
+template(name="outfmt" type="list") {
+  property(name="$!all-json-plain")
+  constant(value="\n")
+}
+
+if $msg contains "msgnum:" then
+ action(
+   type="omfile"
+   file=`echo $RSYSLOG_OUT_LOG`
+   template="outfmt"
+ )
+'
+if [ "x${USE_VALGRIND:-false}" == "xtrue" ] ; then
+	startup_vg
+else
+	startup
+fi
+if [ -n "${USE_GDB:-}" ] ; then
+	echo attach gdb here
+	sleep 54321 || :
+fi
+
+# write the beginning of the file
+echo 'date stdout P msgnum:0' > $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:0"}' > $RSYSLOG_DYNNAME.json.input
+echo 'date stdout F msgnum:1' >> $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:1\n"}' >> $RSYSLOG_DYNNAME.json.input
+echo 'date stdout F msgnum:2' >> $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:2\n"}' >> $RSYSLOG_DYNNAME.json.input
+
+# sleep a little to give rsyslog a chance to begin processing
+if [ -n "${USE_GDB:-}" ] ; then
+	sleep 54321 || :
+else
+	sleep 1
+fi
+
+echo 'date stdout P msgnum:3' >> $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:3"}' >> $RSYSLOG_DYNNAME.json.input
+echo 'date stdout P msgnum:4' >> $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:4"}' >> $RSYSLOG_DYNNAME.json.input
+echo 'date stdout P msgnum:5' >> $RSYSLOG_DYNNAME.crio.input
+echo '{"time":"date", "stream":"stdout", "log":"msgnum:5"}' >> $RSYSLOG_DYNNAME.json.input
+
+# give it time to finish
```

Comment:
if we use this construct more often, it should be moved into the testbench plumbing...